pkg/connectorbuilder/connectorbuilder.go

```diff
@@ -955,34 +976,54 @@ func (b *builderImpl) Revoke(ctx context.Context, request *v2.GrantManagerServic
 
 	l := ctxzap.Extract(ctx)
 
-	rt := request.Grant.Entitlement.Resource.Id.ResourceType
-	provisioner, ok := b.resourceProvisioners[rt]
-	if ok {
-		annos, err := provisioner.Revoke(ctx, request.Grant)
-		if err != nil {
-			l.Error("error: revoke failed", zap.Error(err))
-			b.m.RecordTaskFailure(ctx, tt, b.nowFunc().Sub(start))
-			return nil, fmt.Errorf("error: revoke failed: %w", err)
+	var (
+		attempt   = 0
+		baseDelay = 30 * time.Second
+		rt        = request.Grant.Entitlement.Resource.Id.ResourceType
+	)
+	for {
+		provisioner, ok := b.resourceProvisioners[rt]
```

Comment:
The provisioner won't change. It's probably best to push this out of the for loop.